test/store_unittest.cc

```diff
@@ -0,0 +1,234 @@
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  MetadataStore::Metadata m1(
+      "default-version1",
+      false,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m2(
+      "default-version2",
+      true,
+      std::chrono::system_clock::now(),
+      std::chrono::system_clock::now(),
+      json::object({{"f", json::string("hello")}}));
+  store.UpdateMetadata(mr1, std::move(m1));
+  store.UpdateMetadata(mr2, std::move(m2));
+  const auto metadata_map_before = GetMetadataMap();
+  EXPECT_EQ(2, metadata_map_before.size());
+  EXPECT_EQ("default-version1", metadata_map_before.at(mr1).version);
+  EXPECT_EQ("default-version2", metadata_map_before.at(mr2).version);
+  PurgeDeletedEntries();
+  const auto metadata_map_after = GetMetadataMap();
+  EXPECT_EQ(1, metadata_map_after.size());
+  EXPECT_EQ("default-version1", metadata_map_after.at(mr1).version);
+  EXPECT_THROW(metadata_map_after.at(mr2), std::out_of_range);
+}
+
+TEST(MetadataTest, MetadataParsedCorrectly) {
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::object({{"f", json::string("hello")}}));
+  EXPECT_FALSE(m.ignore);
+  EXPECT_EQ("default-version", m.version);
+  EXPECT_FALSE(m.is_deleted);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+            m.created_at);
+  EXPECT_EQ(time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+            m.collected_at);
+  const json::Object* obj = m.metadata->As<json::Object>();
+  EXPECT_TRUE(obj->Has("f"));
+  EXPECT_EQ("hello", obj->Get<json::String>("f"));
+}
+
+TEST(MetadataTest, MetadataClonedCorrectly) {
+  MetadataStore::Metadata m(
+      "default-version",
+      false,
+      time::rfc3339::FromString("2018-03-03T01:23:45.678901234Z"),
+      time::rfc3339::FromString("2018-03-03T01:32:45.678901234Z"),
+      json::object({{"f", json::string("hello")}}));
+  MetadataStore::Metadata m_clone = m.Clone();
+  EXPECT_EQ(m.version, m_clone.version);
+  EXPECT_FALSE(m_clone.is_deleted);
+  EXPECT_FALSE(m_clone.ignore);
+  EXPECT_EQ(m.created_at, m_clone.created_at);
+  EXPECT_EQ(m.collected_at, m_clone.collected_at);
+  const json::Object* obj = m_clone.metadata->As<json::Object>();
+  EXPECT_TRUE(obj->Has("f"));
+  EXPECT_EQ("hello", obj->Get<json::String>("f"));
+}
+
+TEST(MetadataTest, IgnoredMetadataCorrectlyCreated) {
+  MetadataStore::Metadata m = MetadataStore::Metadata::IGNORED();
+  EXPECT_TRUE(m.ignore);
+}
+
+TEST(MetadataTest, IgnoredMetadataCorrectlyCloned) {
+  MetadataStore::Metadata m = MetadataStore::Metadata::IGNORED();
+  MetadataStore::Metadata m_clone = m.Clone();
+  EXPECT_TRUE(m_clone.ignore);
```

Comment:
[Optional] `EXPECT_EQ(m.ignore, m_clone.ignore)`. But may not be worth it with just one value.

Reply:
Same comment as above.